quiffen/core/account.py

```diff
@@ -101,7 +102,22 @@ def set_header(self, header: AccountType) -> None:
         header : AccountType
             The last header used.
         """
-        self._last_header = header
+        self._last_header = self._convert_deprecated_str_header(header)
+
+    def _convert_deprecated_str_header(self, header):
+        """
+        We used to allow `header` to be a str, but want to enforce it always
+        being of type `AccountType`. This temporary helper function servers to
+        support both for some transition time until we drop it in the next major
+        release.
+        """
+        if not isinstance(header, AccountType):
+            warnings.warn(f'You passed in a value of {header!r} for argument "header", '
+                          f'but it should have been of type "AccountType". Starting with '
+                          f'the next major release, this will become a type error.',
+                          DeprecationWarning)
```

Comment:
Couple of things: `next *minor release`

Secondly, could we format with a hanging indent? :) 

```python
            warnings.warn(
                f'You passed in a value of {header!r} for argument "header", '
                f'but it should have been of type "AccountType". Starting with '
                f'the next major release, this will become a type error.',
                DeprecationWarning,
           )
```

Reply:
Also, the second and third lines don't need to be f-strings

Reply:
Are you looking to follow semver? According to semver, a breaking change requires a major release, which is why I wrote this here. :) Happy to change the copy here, though, if you don't mind.

Reply:
and, changed it to a hanging indent

Reply:
Oh really? I thought a breaking change could have a minor release, and it was just hotfixes that had to be version compatible? Maybe I need to do some reading...